packages/core/src/new-api/internal/execution/transactions/chain-dispatcher.ts

```diff
@@ -2,11 +2,19 @@ import { ethers } from "ethers";
 
 import { GasAdapter, TransactionsAdapter } from "../../../types/adapters";
 
-interface TransactionReceipt {
+interface TransactionSuccess {
+  type: "transaction-success";
   contractAddress?: string;
   txId?: string;
 }
 
+interface TransactionFailure {
+  type: "transaction-failure";
+  error: Error;
```

Comment:
I don't think we should record `Error`s like this, but rather the most low-level representation of the failure that we can get.

In the ase of a transaction we don't get much, as you don't get access to the `returndata` of a failed transaction. We may want to simulate transactions before sending them, and record the return values of those simulations instead.

For static calls this does mean recording the raw data returned by the node we are connected to. It may be simpler to access it by doing the `eth_call` ourselves, and only using ethers for ABI encoding.

Reply:
This has been captured as an issue here: https://github.com/NomicFoundation/ignition/issues/286